.pre-commit-config.yaml

```diff
@@ -0,0 +1,31 @@
+# .pre-commit-config.yaml
+
+# Hooks are divided into active and passive
+# active hooks report and 'fix' things on failure
+# passive hooks only report failure.
+# In both cases, failure of a check will fail an action e.g. commit
+repos:
+- repo: https://github.com/talos-systems/conform
+  rev: master
+  # This hook runs the conform enforcer (passive check)
+  # See .conform.yaml for which rules are enforced
+  hooks:
+  - id: conform
+    stages:
+    - commit-msg
+
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v2.4.0
+  hooks:
+  # This hook trims trailing whitespace (active check).
+  - id: trailing-whitespace
+  # Ensures that a file is either empty, or ends with one newline (active check).
+  - id: end-of-file-fixer
+
+- repo: https://gitlab.com/daverona/pre-commit-cpp
```

Comment:
this does not look right. The goal is to hook commit messages, this also runs clang-format, and ignoring our custom configuration in `.clang-format`.

We already have an action for clang-format on PRs. This should be enough.